tests/Unit/Domain/Test_OrientationMap.cpp

```diff
@@ -259,3 +260,79 @@ SPECTRE_TEST_CASE("Unit.Domain.OrientationMap", "[Domain][Unit]") {
   test_2d();
   test_3d();
 }
+
+SPECTRE_TEST_CASE("Unit.Domain.DiscreteRotation.Rotation", "[Domain][Unit]") {
+  const OrientationMap<1> rotation1(
+      std::array<Direction<1>, 1>{{Direction<1>::lower_xi()}});
+  const std::array<DataVector, 1> test_points1{
+      {DataVector{-1.0, 1.0, 0.7, 0.0}}};
+  const std::array<DataVector, 1> expected_rotated_points1{
+      {DataVector{1.0, -1.0, -0.7, 0.0}}};
+  CHECK(discrete_rotation(rotation1, test_points1) == expected_rotated_points1);
+  CHECK(discrete_rotation(rotation1, std::array<double, 1>{{-0.2}}) ==
+        std::array<double, 1>{{0.2}});
+
+  const OrientationMap<2> rotation2(std::array<Direction<2>, 2>{
+      {Direction<2>::upper_eta(), Direction<2>::upper_xi()}});
+  const std::array<DataVector, 2> test_points2{
+      {DataVector{-1.0, 1.0, 0.7, 0.0}, DataVector{0.25, 1.0, -0.2, 0.0}}};
+  const std::array<DataVector, 2> expected_rotated_points2{
+      {DataVector{0.25, 1.0, -0.2, 0.0}, DataVector{-1.0, 1.0, 0.7, 0.0}}};
+  CHECK(discrete_rotation(rotation2, test_points2) == expected_rotated_points2);
+  CHECK(discrete_rotation(rotation2, std::array<double, 2>{{-1.0, 0.5}}) ==
+        std::array<double, 2>{{0.5, -1.0}});
+
+  const OrientationMap<3> rotation3(std::array<Direction<3>, 3>{
```

Comment:
Can you check all cases, including reversing directions? Try to get as far as you can with for loops, not just writing ~40 or 50 cases out manually because that'll be terrible to maintain and terrible to review.